e2e/tests/init_test.go

```diff
@@ -5,32 +5,52 @@ package e2e
 import (
 	"fmt"
 	"regexp"
+	"strconv"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/regen-network/gocuke"
 	"github.com/stretchr/testify/require"
 )
 
-var addrRe *regexp.Regexp
+var (
+	addrRe        *regexp.Regexp
+	amountRe      *regexp.Regexp
+	accNameToAddr = make(map[string]string)
+	keyRingFlag   = "--keyring-backend=test"
+)
 
 func init() {
-	addrRe = regexp.MustCompile(`address:\s+(pokt1\w+)`)
+	addrRe = regexp.MustCompile(`address: (\S+)\s+name: (\S+)`)
+	amountRe = regexp.MustCompile(`amount: "(.+?)"\s+denom: upokt`)
 }
 
 type suite struct {
 	gocuke.TestingT
-	pocketd *pocketdBin
+	pocketd   *pocketdBin
+	tempState map[string]any // temporary state for each scenario
 }
 
 func (s *suite) Before() {
 	s.pocketd = new(pocketdBin)
+	s.tempState = make(map[string]any)
 }
 
 // TestFeatures runs the e2e tests specified in any .features files in this directory
 // * This test suite assumes that a LocalNet is running
 func TestFeatures(t *testing.T) {
-	gocuke.NewRunner(t, &suite{}).Path("*.feature").Run()
+	runner := gocuke.NewRunner(t, &suite{}).Path("*.feature")
+	runner = registerSteps(runner)
+	runner.Run()
+}
+
+func registerSteps(runner *gocuke.Runner) *gocuke.Runner {
```

Comment:
Do we have to have the `registerSteps` function?

When I played around with gockue in the past, it was able to assume the name of the function based on the syntax of the step

Reply:
So we need this when the gocuke cant figure it out. The methods we register here are due to there being a more complex regex match (ie we want app1 to be matched not hard coded) it can do the more simple ones

Reply:
Give the hard-coding vs regex dillema, do you know how I was able to implement `TheNetworkHasActorsOfType` in [this PR](https://github.com/pokt-network/pocket/pull/874/files) without needing to register a step explicitly?


![Screenshot 2023-10-21 at 12 43 51 PM](https://github.com/pokt-network/poktroll/assets/1892194/efb09a09-2a09-4c8f-abb4-1ff6aeaa2c44)

![Screenshot 2023-10-21 at 12 43 42 PM](https://github.com/pokt-network/poktroll/assets/1892194/ffabc071-5169-408d-ab3a-dd1c4c6979ba)